exporter/chronicleexporter/marshal.go

```diff
@@ -482,26 +519,60 @@ func buildForwarderString(cfg Config) string {
 	return fmt.Sprintf(format, cfg.Project, cfg.Location, cfg.CustomerID, cfg.Forwarder)
 }
 
-func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) map[string]*api.ImportLogsRequest {
-	payloads := make(map[string]*api.ImportLogsRequest, len(rawLogs))
+func (m *protoMarshaler) constructHTTPPayloads(rawLogs map[string][]*api.Log) map[string][]*api.ImportLogsRequest {
+	payloads := make(map[string][]*api.ImportLogsRequest, len(rawLogs))
 
 	for logType, entries := range rawLogs {
 		if len(entries) > 0 {
-			payloads[logType] =
-				&api.ImportLogsRequest{
-					// TODO: Add parent and hint
-					// We don't yet have solid guidance on what these should be
-					Parent: "",
-					Hint:   "",
-
-					Source: &api.ImportLogsRequest_InlineSource{
-						InlineSource: &api.ImportLogsRequest_LogsInlineSource{
-							Forwarder: buildForwarderString(m.cfg),
-							Logs:      entries,
-						},
-					},
-				}
+			request := m.buildHTTPRequest(entries)
+
+			payloads[logType] = m.enforceMaximumsHTTPRequest(request)
 		}
 	}
 	return payloads
 }
+
+func (m *protoMarshaler) enforceMaximumsHTTPRequest(request *api.ImportLogsRequest) []*api.ImportLogsRequest {
+	size := proto.Size(request)
+	logs := request.GetInlineSource().Logs
+	if size > m.cfg.BatchRequestSizeLimitHTTP || len(logs) > m.cfg.BatchLogCountLimitHTTP {
```

Comment:
I'm still working out how size is measured for HTTP requests, but I don't think it's `proto.Size`, and therefore I don't think it's a cheap value to calculate. Therefore, I would suggest handling # of logs per request first, and then checking size afterwards. Splitting logic can be the shared, if the condition can be abstracted.

Reply:
If it's likely not cheap to calculate, maybe it makes sense to do the batch-splitting based on the API response rather than pre-calculating it during payload construction, as we discussed?

Reply:
For posterity, Dan and I discussed this further offline and decided that 1) proto.Size was as reliable as any other measurement method, so with cautious configuration it would work 2) trying to detect when a "too large" error was not feasible with the current state of the v1alpha HTTP API, so any enhancement in that sphere will have to wait for improved API usability.